src/pybind/mgr/cephadm/schedule.py

```diff
@@ -135,67 +135,52 @@ def place(self):
-        else:
-            # exclusive to daemons from 'mon' and 'ha-rgw' services.
-            # Filter out hosts that don't have a public network assigned
-            # or don't allow virtual ips respectively
-            if self.filter_new_host:
-                old = others
-                others = [h for h in others if self.filter_new_host(h.hostname)]
-                for h in list(set(old) - set(others)):
-                    if self.spec.service_type == 'ha-rgw':
-                        logger.info(
-                            f"Filtered out host {h.hostname} for ha-rgw. Could not verify host allowed virtual ips")
-                logger.debug('filtered %s down to %s' % (old, others))
-
-            # ask the scheduler to return a set of hosts with a up to the value of <count>
-            others = self.scheduler.place(others, need)
-            logger.debug('Combine hosts with existing daemons %s + new hosts %s' % (
-                hosts_with_daemons, others))
-            # if a host already has the anticipated daemon, merge it with the candidates
-            # to get a list of HostPlacementSpec that can be deployed on.
-            final_candidates = list(merge_hostspecs(hosts_with_daemons, others))
-
-        return final_candidates
-
-    def get_hosts_with_active_daemon(self, hosts: List[HostPlacementSpec]) -> List[HostPlacementSpec]:
-        active_hosts: List['HostPlacementSpec'] = []
-        for daemon in self.daemons:
-            if daemon.is_active:
-                for h in hosts:
-                    if h.hostname == daemon.hostname:
-                        active_hosts.append(h)
-        # remove duplicates before returning
-        return list(dict.fromkeys(active_hosts))
-
-    def prefer_hosts_with_active_daemons(self, hosts: List[HostPlacementSpec], count: int) -> List[HostPlacementSpec]:
-        # try to prefer host with active daemon if possible
-        active_hosts = self.get_hosts_with_active_daemon(hosts)
-        if len(active_hosts) != 0 and count > 0:
-            for host in active_hosts:
-                hosts.remove(host)
-            if len(active_hosts) >= count:
-                return self.scheduler.place(active_hosts, count)
-            else:
-                return list(merge_hostspecs(self.scheduler.place(active_hosts, count),
-                                            self.scheduler.place(hosts, count - len(active_hosts))))
-        # ask the scheduler to return a set of hosts with a up to the value of <count>
-        return self.scheduler.place(hosts, count)
+        # sort candidates into existing/used slots that already have a
+        # daemon, and others (the rest)
+        existing_active: List[HostPlacementSpec] = []
+        existing_standby: List[HostPlacementSpec] = []
+        others = candidates
+        for d in self.daemons:
+            hs = d.get_host_placement()
+            for i in others:
+                if str(i) == str(hs):
```

Comment:
```suggestion
                if i == hs:

Reply:
i tried that first but it didn't work.  do i need to create a __eq__ method or somethign?

Reply:
works for me:

```python
In [2]: from  ceph.deployment.service_spec import HostPlacementSpec

In [3]: HostPlacementSpec('h', '', '') == HostPlacementSpec('h', '', '')
Out[3]: True

In [4]: HostPlacementSpec('h', '', '') == HostPlacementSpec('other', '', '')
Out[4]: False

In [5]: HostPlacementSpec('h', '', '') == HostPlacementSpec('h', '', 'othername')
Out[5]: False
```

Reply:
hrm yeah me too... i must have been seeing something else when i was testing this early on